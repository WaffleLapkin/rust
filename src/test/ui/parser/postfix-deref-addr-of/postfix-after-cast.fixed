// run-rustfix
#![feature(postfix_addr_of, postfix_deref)]

fn main() {
    _ = (0 as u8).&; //~ error: cast cannot be followed by a postfix address of operator
    _ = (&mut 0u8 as &u8).*; //~ error: cast cannot be followed by a postfix deref
}
