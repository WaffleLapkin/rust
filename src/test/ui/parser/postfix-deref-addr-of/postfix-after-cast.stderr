error: cast cannot be followed by a postfix address of operator
  --> $DIR/postfix-after-cast.rs:5:9
   |
LL |     _ = 0 as u8.&;
   |         ^^^^^^^
   |
help: try surrounding the expression in parentheses
   |
LL |     _ = (0 as u8).&;
   |         +       +

error: cast cannot be followed by a postfix deref
  --> $DIR/postfix-after-cast.rs:6:9
   |
LL |     _ = &mut 0u8 as &u8.*;
   |         ^^^^^^^^^^^^^^^
   |
help: try surrounding the expression in parentheses
   |
LL |     _ = (&mut 0u8 as &u8).*;
   |         +               +

error: aborting due to 2 previous errors

